index.js

```diff
@@ -58,6 +58,55 @@ const validateOptions = ({flags}) => {
 	}
 };
 
+const validateChoicesByFlag = (flagKey, flagValue, receivedInput) => {
+	const {choices, isRequired} = flagValue;
+
+	if (!choices) {
+		return;
+	}
+
+	if (!Array.isArray(choices)) {
+		throw new TypeError('Choices should be array');
+	}
+
+	if (receivedInput === undefined) {
+		if (isRequired) {
+			return `Flag ${flagKey} has no value. Value must be one of: ${choices.join(', ')}`;
+		}
+
+		return;
+	}
+
+	if (Array.isArray(receivedInput)) {
+		const unknownValues = receivedInput.filter(index => !choices.includes(index));
+		if (unknownValues.length === 1) {
+			return `Unknown value: \`${unknownValues[0]}\`. Value must be one of: ${choices.join(', ')}`;
+		}
+
+		if (unknownValues.length > 1) {
+			return `Unknown values: \`${unknownValues.join(', ')}\`. Value must be one of: ${choices.join(', ')}`;
+		}
+	} else if (!choices.includes(receivedInput)) {
+		return `Unknown value: \`${receivedInput}\`. Value must be one of: ${choices.join(', ')}`;
+	}
+};
+
+const validateChoices = (flags, receivedFlags) => {
+	const errors = [];
+
+	for (const [flagKey, flagValue] of Object.entries(flags)) {
+		const receivedInput = receivedFlags[flagKey];
+		const errorMessage = validateChoicesByFlag(flagKey, flagValue, receivedInput);
+		if (errorMessage) {
+			errors.push(errorMessage);
+		}
+	}
+
+	if (errors.length > 0) {
+		throw new Error(`${errors.join('. ')}`);
```

Comment:
Please add test for case of multiple errors.

Reply:
@sindresorhus The output from here will be:

```
Unknown values: `dog, rabbit`. Value must be one of: cat, unicorn. Unknown values: `dog, rabbit`. Value must be one of: cat, unicorn. Unknown values: `dog, rabbit`. Value must be one of: cat, unicorn
```

Is this acceptable for you?

Reply:
We are using `new Error()` for throwing error to developers. But this error message is for users. We should use `console.error` instead.